api/tests/unit/domain/usecases/authenticate-external-user_test.js

```diff
@@ -87,6 +88,44 @@ describe('Unit | Application | UseCase | authenticate-external-user', function()
 
     });
 
+    it('should save last logged at date when authentication succeeds', async function() {
+      // given
+      const password = 'Azerty123*';
+      const user = createUserWithValidCredentials({
+        password,
+        authenticationService,
+        userRepository,
+      });
+
+      const externalUserToken = 'external user token';
+      createValidConditionsForAddingGarAuthenticationMethod({
```

Comment:
**chipotage**

Concernant ce helper existant (introduit il y a quelques mois).

Ce que fait ce code est:
- renvoyer le samlId depuis le token
- empêcher de renvoyer l'erreur "autre utilisateur déjà réconcilié sur même samlId"
- ne rejette pas la création de la méthode d'authentification 

Du coup, j'ai envie de 
- renommer en `appleSauce` ou `setup` ou le mettre dans un beforeEach
- remettre le code dans chaque test

```suggestion
      createValidConditionsForAddingGarAuthenticationMethod({
```

Je sais, réintroduire de la duplication, c'est pas naturel, mais si on arrive pas à nommer c'est qu'il y a un truc
> duplication is far cheaper than the wrong abstraction
https://sandimetz.com/blog/2016/1/20/the-wrong-abstraction


Dans tous les cas, on peut simplifier la méthode en mettant le cas passant (samlId jamais utilisé): modifier en `userRepository.getBySamlId.withArgs(samlId).resolves();` (et en conséquence enlever le paramètre user).